src/scraper.rs

```diff
@@ -313,29 +337,67 @@ impl Scraper {
         std::thread::sleep(delay_duration);
     }
 
-    /// If a URL should be visited, or does it belong to another domain
-    fn should_visit(url: &str, base: &Url) -> bool {
+    /// If a URL should be visited (ignores `mail:`, `javascript:` and other pseudo-links)
+    fn should_visit(url: &str) -> bool {
         match Url::parse(url) {
+            /* The given candidate is a valid URL, and not a relative path to
+             * the next one. Therefore, we have to check if this URL is valid.
+             * If it is, we should visit it.
+             */
+            Ok(not_ok) => not_ok.has_host() && !not_ok.cannot_be_a_base(),
+
+            /* Since we couldn't parse this "URL", then it must be a relative
+             * path or a malformed URL. If the URL is malformed, then it will
+             * be handled during the join() call in run() */
+            Err(_) => true,
+        }
+    }
+
+    /// Replaces `///` with `//`
+    /// And `//` with `https://`
+    /// Without this function, if url is `///tools.wmflabs.org/`, the app crashes.
```

Comment:
```suggestion
    /// Without this function, if url is `///<domain>.<extension>/`, the app crashes.
```
I think this is better. Otherwise, the comment makes it sound like using tools.wmflabs.org was the only buggy URL